### ISSUE TITLE
Private events lose their private status when sold out!

### ISSUE THREAD

**Issue by jonappleton** [eventespresso/event-espresso-core#278]
> Before submitting your issue, please ensure that what you are reporting is specific to Event Espresso 4.
Yes EE Version 4.9.46.p

When submitting an issue, please follow this rough guideline for your report.

### Is this a bug report?

* [ X] yes
* [ ] no

### What branch did this occur in?
Version 4.9.46.p

### What version of WordPress is on the site this issue happened?
4.8.5

### If you were directed to create an issue here by the EE support team, what is the link to your original EE support forum thread?
https://eventespresso.com/topic/private-events-lose-their-private-status-when-sold-out/#post-263807

### Outline the steps to reproduce
Create and event
Set it to private (which shows in EE admin as Inactive)
Set it to have 1 ticket qty for sale
Register someone for that ticket so qty left becomes zero
EE then changes the post_status of that event from "private" to "sold_out"
So now the event shows as being public and no longer private

We have an upcoming schedule table that shows events where post_status = published or sold_out as we still like to show the sold out events as those booked on to them can click to see location details etc.

We’ve just noticed that when the last ticket sells on a private event then EE changes the post_status from private to sold_out, causing our private events to show in our public website schedule.

I did some digging and found that EE stores the old post_status in the postmeta table with a key of “_previous_event_status” so I have now added an extra check to my php so when is iterates through the results it says

if($previous_event_status == “private” && $currentEventStatus == “sold_out”) {

//dont show table row } else {
//yes display this event
}

This seems to work for me but just wondering if I’m making extra work for myself and is there a simpler way to ignore/hide private events when pulling events from SQL

This feels like a bit of a bug that EE overrides the private status like this?

Thanks
Jonathan

**Comment by mnelson4** [comment]
hey @jonappleton ya I'm reproducing the issue. When events get sold out, they're given a new status of "sold_out"... which is public! D'oh! It seems we should have a "sold_out_private" and "sold_out_protected" too.
I'm going to see if we already have an internal ticket for this...

**Comment by jonappleton** [comment]
Thanks for checking

**Comment by mnelson4** [comment]
Hey @jonappleton believe it or not, this is the first report of this issue. (I guess not a lot of people use protected/private events that get sold out). We're going to work on a quickfix solution and a more long-term solution.
For now, you could increase the ticket limit on the sold-out event, (so that it won't be sold out anymore) and remove the ticket selector from the event (so people won't be able to register anymore). This is obviously not the ideal solution, but it will make it so these events won't be public anymore, but will also not allow new registrations

**Comment by jonappleton** [comment]
Hi, Yes we can work with that as these private events should only be managed by site admins and registrations manually added in backend so Ill ask them for now to set the QTY available to be higher than the actual max they go up to so it never hits sold out :)

**Comment by tn3rb** [comment]
internal ticket: https://events.codebasehq.com/projects/event-espresso/tickets/11392

**Comment by mnelson4** [comment]
in the REST API, on pull request https://github.com/eventespresso/event-espresso-core/pull/536, we made use of the postmeta `_previous_event_status` to determine if the sold out event was PREVIOUSLY public. IMO we should probably do something similar on the frontend

**Comment by aaron-price** [comment]
I'm having an issue related to this. I'm accessing the events via the REST API, but after an event is sold out, it's no longer available. Trying to GET it directly results in

```
{
    "code": "rest_user_cannot_read",
    "message": "Sorry, you cannot read this event. Missing permissions are: ee_read_events,ee_read_others_events,ee_read_private_events",
    "data": {
        "status": 403
    }
}
```

**Comment by garthkoyle** [comment]
@sethshoultes do you want this still open?

**Comment by joshfeck** [comment]
@mnelson4 or @tn3rb 

How much effort would it take to add a check for the private post status and bail before changing the post status to sold out?

**Comment by mnelson4** [comment]
I suspect Brent would know better, but I would guess that would be pretty easy. But I'm not sure what the side-effects would be to having an event that's effectively sold-out, but its status hasn't been updated to that.

**Comment by joshfeck** [comment]
I don't think there would be any side effects because the ticket selector would continue to say "Sold Out".

**Comment by mnelson4** [comment]
Well, I tried that on https://github.com/eventespresso/event-espresso-core/pull/960 and ya, so far I haven't noticed any harm in it. The sold-out and private events are instead just marked "inactive", which is kinda true.

**Comment by joshfeck** [comment]
Fixed in #960
